aurelia-slickgrid/src/aurelia-slickgrid/editors/dateEditor.ts

```diff
@@ -91,14 +91,22 @@ export class DateEditor implements Editor {
   }
 
   serializeValue() {
+    const domValue: string = this.$input.val();
+
+    if (!domValue) return '';
+
     const outputFormat = mapMomentDateFormatWithFieldType(this.args.column.type || FieldType.dateIso);
-    const value = moment(this.defaultDate).format(outputFormat);
+    const value = moment(domValue).format(outputFormat);
 
     return value;
   }
 
   applyValue(item: any, state: any) {
-    item[this.args.column.field] = state;
```

Comment:
hmm is that necessary? I thought the `state` was already formatted by the serializer

Reply:
i guess it depends how you want to handle this. Currently `serailizedValue` method returns a string date using `moment(date).format(string)`. If you used just the `state` value in `applyValue` then the original property for this editor will go from a date object type to a string type, and I do not think you want that? If you do not want to serialize it as a string, then this logic can move into the `serializedValue` method so the return value is a date object and then `applyValue` will go back to `item[this.args.column.field] = state;`

Reply:
ah right I didn't think about that, it is a string, I guess your change is fine then

Reply:
so do you think it is right to serialize the date as a string and then deserilize in `applyValue`. The only downside I can see with that is that if someone overrides the `editCommandHandler`, the serializedValue will be a string and not a date object. Per the slickgrid documents he describes the function of `serializedValue` as:

> 
> // return the value(s) being edited by the user in a serialized form
>         // can be an arbitrary object
>         // the only restriction is that it must be a simple object that can be passed around even
>         // when the editor itself has been destroyed

And i think a date is a simple object that can be passed around after the editor is destroyed right?

Reply:
you can add a variable in the `onChange` to keep actual date and if never changed then just return the `defaultDate`. So something like this 
`return this.currentDate || this.defaultDate;`

Reply:
We need to make the Excel Copy Buffer Plugin working before we can say that the Editor works correctly

Reply:
I got it to work by adding `this.flatInstance.setDate(item[this.args.column.field])` in `loadValue` right after you set the `defaultDate`. However, I had a few questions before i commit my changes:

1. Is there a reason why you do not destroy flatpickr? This was commented out and instead you close it. However, it seems on init you are resetting the instance with a new element anyway.
2. You check for flatpickr with this code:

`this.flatInstance = (flatpickr && this.$input[0] && typeof this.$input[0].flatpickr === 'function') ? this.$input[0].flatpickr(pickerOptions) : null;`

Was there a reason you did not import flatpickr like `import flatpickr from 'flatpickr` and then initialize it like `this.flatInstance = flatpickr ? flatpickr(this.$input.get(0), pickerOptions) : null;`. In their documents it says this is the preferred way if their library is in a framework.

Reply:
The `flatpick` library is not that great with how to import it in TS, they changed it a few times too in trying to make it work in TS. Also it was the only way I got it working with both `WebPack` and RequireJS (the client-cli demo I have). So please don't change the import, it works now, don't break it again lol

I think I removed the delete of the `flatInstance` because SlickGrid destroys the DOM element even before `flatInstance`. Might not be exactly that, but it was something near that behavior.

Reply:
thanks for that clarification, I keep forgetting about the CLI/RequireJS. I pushed the commit that gets the date to work with the excel cell copy